packages/perseus-core/src/analytics.ts

```diff
@@ -1,12 +1,25 @@
 // A type union of all the events that any package in the Perseus ecosystem can
 // send.
-export type PerseusAnalyticsEvent = {
-    type: "perseus:expression-evaluated";
-    payload: {
-        virtualKeypadVersion: string;
-        result: "correct" | "incorrect" | "invalid";
-    };
-};
+export type PerseusAnalyticsEvent =
+    | {
+          type: "perseus:expression-evaluated";
+          payload: {
+              virtualKeypadVersion: string;
+              result: "correct" | "incorrect" | "invalid";
+          };
+      }
+    | {
+          type: "perseus:keypad-closed";
+          payload: {
+              virtualKeypadVersion: string;
+          };
+      }
+    | {
+          type: "perseus:keypad-opened";
+          payload: {
+              virtualKeypadVersion: string;
+          };
+      };
```

Comment:
As per a previous comment, let's prefix these new events with `math-input:` instead of `perseus:` just so events are easy to differentiate. I'm unsure at this point if this is necessary, but I think it's worth it for now.